### PR TITLE
Enabling pipeline E2E tests for dual stack

### PR DESCRIPTION
**Pull request by RLeclair**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

Adding two jobs to test E2E functionality of new `c-utility` dual stack changes. Testing on both Ubuntu 20.04 and Ubuntu 22.04.

### CODE REVIEW

File: build/.vsts-ci.yml [hunk 1]
```diff
@@ -508,6 +508,94 @@ jobs:
       mergeTestResults: true
       testRunTitle: 'ubuntuRIOT'
     condition: succeededOrFailed()
+- job: ubuntu2004dualstack
+  container: linux-c-ubuntu-2004
+  pool:
+    name: 'sdk-c--ubuntu-18'
```
Comment (vaavva) [inline build/.vsts-ci.yml:514]: Should this be 2004?

File: build/.vsts-ci.yml [hunk 2]
```diff
@@ -508,6 +508,94 @@ jobs:
...
+    inputs:
+      testRunner: CTest
+      testResultsFiles: '**/Test.xml'
+      mergeTestResults: true
+      testRunTitle: 'ubuntu2004dualstack'
+    condition: succeededOrFailed()
+- job: ubuntu2204dualstack
+  container: linux-c-ubuntu-2204
+  pool:
+    name: 'sdk-c--ubuntu-18'
```
Comment (ewertons) [inline build/.vsts-ci.yml:556]: This also is not matching the title of the job.
Is it supposed to be ubuntu 22.04 ?

**Review by ewertons — CHANGES_REQUESTED** [review]

File: build/.vsts-ci.yml [hunk 3]
```diff
@@ -508,6 +508,94 @@ jobs:
       mergeTestResults: true
       testRunTitle: 'ubuntuRIOT'
     condition: succeededOrFailed()
+- job: ubuntu2004dualstack
+  container: linux-c-ubuntu-2004
+  pool:
+    name: 'sdk-c--ubuntu-18'
+  displayName: 'Ubuntu 20.04 with Dual Stack'
```
Comment (ericwol-msft) [inline build/.vsts-ci.yml:515]:
```suggestion
  displayName: 'Ubuntu 20.04 with Dual IP Stack'
```

File: build/.vsts-ci.yml (hunk 2 above)
Comment (ericwol-msft) [reply build/.vsts-ci.yml:556]: that is the pool name. Is 22 supported in 1es yet?

Comment (ericwol-msft) [reply build/.vsts-ci.yml:556]: https://learn.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software

Comment (ericwol-msft) [reply build/.vsts-ci.yml:556]: ubuntu-22.04

**Review by ewertons — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

**Pull request merged**